feat: convert collaboration cycle team field to dropdown

humancompanion-usds opened the pull request:
## Summary

This PR converts the **VFS team name** field in the Collaboration Cycle request template from a text input to a dropdown menu with all current teams.

## Changes

- ✅ Converted  field from  to 
- ✅ Populated with 60 teams in format: "Team Name (12345)"
- ✅ Teams sorted alphabetically for easy selection
- ✅ Added helpful description: "Select your team from the list (sorted alphabetically)"

## Benefits

1. **Better UX** - Easy to find and select the correct team
2. **Data consistency** - Prevents typos and variations in team names
3. **Easier tracking** - Team IDs make it simple to match requests to teams
4. **Future-proof** - Dropdown will be automatically maintained by automation

## Automation Note

Going forward, this dropdown will be automatically updated when team data changes, via the automation in PR #127758. This PR gives us the immediate benefit of the dropdown while the automation is being reviewed.

## Testing

Create a new Collaboration Cycle request issue and verify:
- [ ] Team dropdown appears with all 60 teams
- [ ] Teams are sorted alphabetically
- [ ] Format is "Team Name (ID)"
- [ ] Field is marked as required

---

Related: #127758 (automation PR)

humancompanion-usds committed:
feat: convert collaboration cycle team field to dropdown

Converted the VFS team name field from a text input to a dropdown
with all current teams sorted alphabetically.

Teams are displayed in the format 'Team Name (12345)' with their
unique team ID for easy identification.

This provides a better user experience by:
- Preventing typos and inconsistent team names
- Making it easier to find and select the correct team
- Ensuring data consistency across collaboration cycle requests

Total teams: 60

The dropdown will be automatically maintained going forward via
automation when team data is synced from va.gov-team-sensitive.

humancompanion-usds committed:
fix: remove invalid placeholder attribute from dropdown field

Dropdown fields don't support the placeholder attribute - only input
fields do. This was causing a validation error in GitHub's template
parser.

erinrwhite commented on .github/ISSUE_TEMPLATE/collaboration-cycle-request.yml:
```diff
@@ -1,223 +1,314 @@
+  id: product-team
+  attributes:
+    label: VFS team name
+    description: Select your team from the list (sorted alphabetically)
```
We should have a path here for teams who can't find themselves on the list. I recommend adding something like, 

> Not seeing your team here? [Add your team to the manifest.](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/teams-manifest-user-guide.md#adding-a-new-team)

humancompanion-usds committed:
Enhance dropdown description for VFS team selection

Updated the description for the VFS team name dropdown to include a link for adding teams to the manifest.

allison0034 reviewed (commented):
Looks good!
